optuna/_convert_positional_args.py

```diff
@@ -48,8 +50,8 @@ def convert_positional_args(
     *,
     previous_positional_arg_names: Sequence[str],
     warning_stacklevel: int = 2,
-    deprecated_version: str | None = None,
-    removed_version: str | None = None,
+    deprecated_version: Optional[str] = None,
+    removed_version: Optional[str] = None,
```

Comment:
Could you revert here?
It is recommended to use `str | None` instead of `Optional[str]` according to [PEP604](https://peps.python.org/pep-0604/).